extensions/libraries/redcore/translation/helper.php

```diff
@@ -789,4 +789,142 @@ public static  function setLanguage($language, $loadLanguage = true)
 
 		return false;
 	}
+
+	/**
+	 * Checks if the current page is a translatable form.
+	 *
+	 * @return void
+	 */
+	public static function isTranslatableForm()
+	{
+		$input = JFactory::getApplication()->input;
+		$option = $input->getString('option', '');
+		$view = $input->getString('view', '');
+		$layout = $input->getString('layout', '');
+		$task = $input->getString('task', '');
+
+		$translationTables = self::getInstalledTranslationTables();
+
+		if ($layout == 'edit' || $task == 'edit')
+		{
+			//Go through all installed translation tables
+			foreach ($translationTables as $tableKey => $translationTable)
+			{
+				if (!empty($translationTable->formLinks))
+				{
+					//Go through all form links
+					foreach ($translationTable->formLinks as $formLink)
+					{
+						$formLinks = explode('#', $formLink);
+
+						//Check whether the form link values matches the current page
+						if ($formLinks[0] == $option && $formLinks[1] == $view)
+						{
+							//Get id of item based on the name of the primary key gotten from the XML file.
+							$itemid = $input->getInt($formLinks[2], '');
+
+							//If the item doesn't have an ID, tell the user that they have to save the item first.
+							if (empty($itemid))
+							{
+								self::renderTranslationModal(false, false, false);
+								return;
+							}
+
+							if ($option == 'com_reditem')
```

Comment:
Right. I think I took something you said at some point too literal, which resulted in this. It always bothered me, so I'm actually glad to see that it should be changed.